website/docs/guides/data-interactions/queries.mdx

```diff
@@ -152,6 +152,284 @@ query ($id: ID!) {
+
+```graph
+type Posts @createModel(accountRelation: LIST, description: "A simple Post")
+@createIndex(fields: [{path: "title"}])
+@createIndex(fields: [{path: "tag"}])
+@createIndex(fields: [{path: "ranking"}])
+@createIndex(fields: [{path: "created_at"}])
+{
+    title: String! @string(minLength: 1, maxLength: 100)
+    body: String! @string(minLength: 1, maxLength: 100)
+    tag: String! @string(minLength: 1, maxLength: 100)
+    ranking: Int!
+    created_at: DateTime!
+}
+```
+
+We can then create some example data with different inputs by running a series of [mutations](./mutations.mdx#create-data) like the following:
+
+
+*Query to create a post:*
+```graphql
+mutation CreatePosts($input: CreatePostsInput!){
+  createPosts(input: $input){
+    document{
+      title
+      body
+      tag
+      ranking
+      created_at
+    }
+  }
+}
+```
+
+
+*Input variable with content for the post:*
+```json
+{
+  "input": {
+    "content": {
+      "title": "Let's talk about web3",
+      "body": "This is my first tutorial covering web3",
+      "tag": "tutorial",
+      "ranking": 10,
+      "created_at": "2023-07-11T10:02:08.190Z"
+    }
+  }
+}
+```
+
+With indices defined and some data available we can now perform various queries with filters applied to specific fields.
+
+#### Filter documents based on value conditions
+
+ComposeDB enables you to filter documents based on specific [value conditions](../../api/runtime/schema.mdx#value-conditions) applied to a specific field. Below is an example query that filters posts based on the post ranking. Key things to understand about the query in the example below:
+
+- variable `input` of a type `PostsFiltersInput` allows the `postsIndex` to be filtered by a specified condition
+- the `input` variable below defines the condition that is used to filter the documents. In this specific example, the query will return the documents that have a ranking greater than `8`.
+
+*Query:*
+
+```graphql
+query MyQuery($input:PostsFiltersInput!) {
+  postsIndex(filters:$input, first:5){
+    edges{
+      node{
+        title
+        body
+        created_at
+		ranking
+		tag
+      }
+    }
+  }
+}
+```
+
+*Filtering condition:*
+
+```json
+{
+  "input":{
+    "where": {
+      "ranking":{
+        "greaterThan": 8
+      }
+    }
+  }
+}
+
+```
+
+Let’s look at a few more examples of what queries with value conditions you can write. We will reuse the same query called `MyQuery` as above and will only modify the query conditions.
+
+Return all documents where the field `tag` is not null:
+
+```json
+{
+  "input":{
+    "where": {
+      "tag":{
+        "isNull": false
+      }
+    }
+  }
+}
+
+```
+
+Return all documents where the field tag has a value `tutorial` or `guide`:
+
+```json
+{
+  "input":{
+    "where": {
+      "tag":{
+        "in": ["tutorial", "guide"]
+      }
+    }
+  }
+}
+```
+
+Return documents that were created after a specific date:
+
+```json
+{
+   "input":{
+         "where":{
```

Comment:
The code varies between 2 spaces and 4 spaces. Can we change all code snippets to be 2 spaces for readability?